tests/integration_tests.py

```diff
@@ -0,0 +1,139 @@
+License: Apache 2.0
+Description: Integration tests for coconut
+"""
+
+from __future__ import print_function, absolute_import, unicode_literals, division
+
+from coconut.root import *
+
+import unittest
+import sys
+import os
+import subprocess
+import shutil
+import functools
+
+call = functools.partial(subprocess.check_call,stderr=sys.stdout.fileno())
+
+class OldCocoTest(unittest.TestCase):
+
+    def _compile_extras(self,extraCommands = []):
+        src = os.path.join(self.src,"extras.coco")
+
+        cmd = ["coconut"] + extraCommands + [src,self.bin]
+        call(cmd)
+
+    def _compile_runner(self,extraCommands = []):
+        src = os.path.join(self.src,"src","runner.coco")
+
+        cmd = ["coconut"] + extraCommands + [src,self.bin]
+        call(cmd)
+
+    def _compile_agnostic(self,extraCommands = []):
+        src = os.path.join(self.src,"src","agnostic")
+        bin = os.path.join(self.bin,"cocotest")
+
+        cmd = ["coconut"] + extraCommands + [src,bin]
+        call(cmd)
+
+    def _compile_2(self,extraCommands = []):
+        src = os.path.join(self.src,"src","python2")
+        bin = os.path.join(self.bin,"cocotest")
+
+        cmd = ["coconut","--target","2"] + extraCommands + [src,bin]
+        call(cmd)
+
+    def _compile_3(self,extraCommands = []):
+        src = os.path.join(self.src,"src","python3")
+        bin = os.path.join(self.bin,"cocotest")
+
+        cmd = ["coconut","--target","3"] + extraCommands + [src,bin]
+        call(cmd)
+
+    def _compile_35(self,extraCommands = []):
+        src = os.path.join(self.src,"src","python35")
+        bin = os.path.join(self.bin,"cocotest")
+
+        cmd = ["coconut","--target","35"] + extraCommands + [src,bin]
+        call(cmd)
+
+    def _run_source(self):
+        call(["python",os.path.join(self.bin,"runner.py")])
+
+    def _compile_source(self,agnosticTarget = None, stict = False, minify = False, line_numbers = False, keep_lines = False):
+
+        agnosticCommands = []
+        extraCommands = []
+        if agnosticTarget != None:
+            agnosticCommands += ["--target",agnosticTarget]
+        if stict:
+            extraCommands += ["--strict"]
+        if line_numbers:
+            extraCommands += ["--line-numbers"]
+        if keep_lines:
+            extraCommands += ["--keep-lines"]
+
+        self._compile_runner(extraCommands+agnosticCommands)
+        self._compile_agnostic(extraCommands+agnosticCommands)
+        
+        if sys.version_info >= (3,):
+            self._compile_3(extraCommands)
+        if sys.version_info >= (3,5):
+            self._compile_35(extraCommands)
+        if (sys.version_info >= (2,)) & (sys.version_info < (3,)):
+            self._compile_2(extraCommands)
+
+    def _clean(self):
+        shutil.rmtree(self.bin)
+
+    def setUp(self):
+        self.src = os.path.join(os.path.abspath(os.path.dirname(__file__)),"src")
+        self.bin = os.path.join(os.path.abspath(os.path.dirname(__file__)),"bin")
```

Comment:
`bin` is usually reserved for a folder containing binary files, can you rename this to `dest`?